### ISSUE TITLE
Considering the use of ES5 accessors

### ISSUE THREAD

**Issue by jmvtrinidad** [knockout/tko.observable#1]
I find this [ES5 plugin](https://github.com/SteveSanderson/knockout-es5) awesome working with typescript and knockout. In terms of development I find writing myself Plain Javascript when using this one. Could you consider this to support it by default?

**Comment by brianmhunt** [comment]
Thanks for the feedback.

There are some issues with the ES5 plugin, but the idea behind it is great.

Another cool option will be ES6 Proxies.

Will mull this. 😀

**Comment by jmvtrinidad** [comment]
In ES6 Proxies, what would be the best way for the developers to interact with it?
Is it something like this one?

```
function Person(){
  this.name = 'Sample'
  this.age = 15  
  //Option 1
  return new Proxy(this, makeObservableHandler)
}
//Option 2
var personInstance = new Proxy(new Person(), makeObservableHandler)

```

Another cool option will be [knockout-decorator](https://github.com/variotry/knockout-decorator). 

And also can you point me to issue that you mean on ES5 plugin?

**Comment by brianmhunt** [comment]
Thanks re. knockout-decorator.  Didn't know about that.

ES6 proxies would look something like that -- except they'd trigger dependency checking and potentially update automatically (computeds).  We'd need a whole new and thorough API for that, and since proxies cannot be shimmed and are poorly supported right now it's unlikely to make it into a mainline design anytime soon.

Re. ES5 - There are a few on the knockout/knockout issue tracker e.g. https://github.com/knockout/knockout/issues/1789

**Comment by jmvtrinidad** [comment]
I don't think it is an issue(#[1789](https://github.com/knockout/knockout/issues/1789)). In order to pass observable in html binding with nested component or plugins, developers need to use `ko.getObservable($data, 'propertyName')`. 

I made a [pullRequest](https://github.com/SteveSanderson/knockout-es5/pull/60) in order for `ko.getObservable` to work with knockout-es5 property and normal declared observables.

Also notice this [example](https://jsfiddle.net/jmvtrinidad/L4o9emjq/) when passing property name in component's param, knockout evaluates it as a computedObservable. But when passed it using `ko.getObservable` it becomes observable.
